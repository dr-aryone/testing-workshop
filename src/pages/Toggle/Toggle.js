import React, { useState } from "react";
import Title from "../../components/Title/Title";

const Toggle = () => {
  const [show, toggleShow] = useState(false);

  return (
    <div data-testid="toggleContainer">
      <Title title="Toggle" />
      <button onClick={() => toggleShow(!show)} data-testid="toggleButton">
        Toggle Paragraph
      </button>
      {show && (
        <p data-testid="toggleParagraph">
          Lorem ipsum dolor sit amet consectetur adipisicing elit. Mollitia est
          facere veniam vel hic nemo quaerat doloremque iste sint corrupti?
          Totam voluptatem consequatur sed vel enim repudiandae ad dolor
          facilis!
        </p>
      )}
    </div>
  );
};

export default Toggle;
