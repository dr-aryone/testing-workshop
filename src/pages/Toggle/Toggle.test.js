import React from "react";
import ReactDOM from "react-dom";
import { render, fireEvent } from "@testing-library/react";
import Toggle from "./Toggle";

describe("Toggle Page", () => {
  it("Renders without crashing", () => {
    ReactDOM.render(<Toggle />, document.createElement("div"));

    ReactDOM.unmountComponentAtNode(document.createElement("div"));
  });

  it("Loads component", () => {
    const { getByTestId } = render(<Toggle />);

    const toggle = getByTestId("toggleContainer");

    expect(toggle).toBeInTheDocument();
  });

  it("Match Snapshot", () => {
    const { firstChild } = render(<Toggle />);

    expect(firstChild).toMatchSnapshot();
  });

  it("Toggle paragraph", () => {
    const { getByText, queryByTestId } = render(<Toggle />);

    const toggleButton = getByText("Toggle Paragraph");

    fireEvent.click(toggleButton);

    const visibleToggleParagraph = queryByTestId("toggleParagraph");

    expect(visibleToggleParagraph).toBeInTheDocument();

    fireEvent.click(toggleButton);

    const notVisibleToggleParagraph = queryByTestId("toggleParagraph");

    expect(notVisibleToggleParagraph).not.toBeInTheDocument();
  });
});
