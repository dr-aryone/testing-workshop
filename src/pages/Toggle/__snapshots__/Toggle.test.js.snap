// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Toggle Page Match Snapshot 1`] = `undefined`;
